cli/pcluster/config/validators.py

```diff
@@ -1423,3 +1411,58 @@ def validate_backup_options(
         fsx_imported_file_chunk_size or fsx_import_path or fsx_export_path or fsx_auto_import_policy
     ) and fsx_automatic_backup_retention_days:
         errors.append("Backups cannot be created on S3-linked file systems")
+
+
+def ebs_volume_size_snapshot_validator(section_key, section_label, pcluster_config):
+    """
+    Validate the following cases.
+
+    The EBS snapshot is in "completed" state if it is specified
+    If user specified the volume size, the volume must be larger than then volume size of the EBS snapshot
+    """
+    errors = []
+    warnings = []
+
+    section = pcluster_config.get_section(section_key, section_label)
+    if section.get_param_value("ebs_snapshot_id"):
+        try:
+            ebs_snapshot_id = section.get_param_value("ebs_snapshot_id")
+            snapshot_response_dict = get_ebs_snapshot_info(ebs_snapshot_id, raise_exceptions=True)
+            # validate that the input volume size is larger than the volume size of the EBS snapshot
+            snapshot_volume_size = snapshot_response_dict.get("VolumeSize")
+            volume_size = section.get_param_value("volume_size")
+            if volume_size < snapshot_volume_size:
+                errors.append("The EBS volume size must not be smaller than the volume size of EBS snapshot.")
+            # validate that the state of ebs snapshot
+            if snapshot_response_dict.get("State") != "completed":
+                warnings.append(
+                    "Snapshot {0} is in state '{1}' not 'completed'".format(
+                        ebs_snapshot_id, snapshot_response_dict.get("State")
+                    )
+                )
+            if volume_size > snapshot_volume_size:
```

Comment:
minor: i would move this check underneath the check for `volume_size < snapshot_volume_size`, and I would use an if-elif for it. The idea is to show that the checks are related (and both of them can't be true at the same time).

```python
if volume_size < snapshot_volume_size:
  # append error
elif volume_size > snapshot_volume_size:
  # append warning
```

Reply:
Done!